src/sessions.ts

```diff
@@ -703,11 +704,11 @@ function endTransaction(
   }
 
   if (txnState === TxnState.TRANSACTION_COMMITTED) {
-    writeConcern = Object.assign({ wtimeout: 10000 }, writeConcern, { w: 'majority' });
+    writeConcern = Object.assign({ wtimeoutMS: 10000 }, writeConcern, { w: 'majority' });
   }
 
   if (writeConcern) {
-    Object.assign(command, { writeConcern });
+    WriteConcern.apply(command, writeConcern);
   }
 
   if (commandName === 'commitTransaction' && session.transaction.options.maxTimeMS) {
```

Comment:
On line 774 in the retryable write handling there's more WriteConcern applying

Reply:
Changed to use apply.